src/awkward/_nplikes/array_module.py

```diff
@@ -97,8 +97,10 @@ def meshgrid(
 
     def array_equal(
         self, x1: ArrayLike, x2: ArrayLike, *, equal_nan: bool = False
-    ) -> bool:
-        return self._module.array_equal(x1, x2, equal_nan=equal_nan)
+    ) -> ArrayLike:
+        return self._module.asarray(
+            self._module.array_equal(x1, x2, equal_nan=equal_nan)
+        )
```

Comment:
We have a policy that typetracer should not impose assumptions about concrete operations. I noticed this here, and changed it.